src/node/utils/customError.js

```diff
@@ -1,17 +1,23 @@
-/*
- This helper modules allows us to create different type of errors we can throw
-*/
-function customError(message, errorName)
-{
-  this.name = errorName || "Error";
-  this.message = message;
-
-  var stackParts = new Error().stack.split("\n");
-  stackParts.splice(0,2);
-  stackParts.unshift(this.name + ": " + message);
-
-  this.stack = stackParts.join("\n");
+/**
+ * CustomError
+ *
+ * This helper modules allows us to create different type of errors we can throw
+ *
+ * @class CustomError
+ * @extends {Error}
+ */
+class CustomError extends Error {
+  /**
+   * Creates an instance of CustomError.
+   * @param {*} message
+   * @param {string} [name='Error'] a custom name for the error object
+   * @memberof CustomError
+   */
+  constructor(message, name = 'Error') {
+    super(message);
+    this.name = name;
+    Error.captureStackTrace(this, this.constructor);
+  }
 }
-customError.prototype = Error.prototype;
 
-module.exports = customError;
+module.exports = CustomError;
```

Comment:
This case change in an exported symbol is a change in the module's API, isn't it?

I think this could bite us later on.

Reply:
Wait, how would this change the module’s API?

Reply:
Nevermind. That was a nonsense.